docker-compose.yml

```
@@ -1,5 +1,16 @@
 version: '3'
 services:
+  nginx:
+    image: nginx
+    ports:
+      - 80:80
```

Once in a while, port 80 on the docker host may be occupied, allowing an override by env var might be helpful?
```suggestion
      - ${STAC_FASTAPI_NGINX_PORT:-80}:80
```

---

does it mean you also need to change https://github.com/stac-utils/stac-fastapi/blob/17305101bd656e2d730affc983a815db31c01541/nginx.conf#L5 ?

---

No, the first number is the port to bind to on the docker host, the second one the port in the container, so that would still be Nginx's port 80.

One potential gotcha is when full URLs are built in the actual service, it needs a way to figure out the external port.